caddyhttp/staticfiles/fileserver.go

```diff
@@ -40,6 +40,14 @@ func (fs FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, err
 	return fs.serveFile(w, r, r.URL.Path)
 }
 
+// calculateEtag produces a strong etag by default. Prefix the result with "W/" to convert this into a weak one.
+// see https://tools.ietf.org/html/rfc7232#section-2.3
+func calculateEtag(d os.FileInfo) string {
+	t := strconv.FormatInt(d.ModTime().Unix(), 36)
+	s := strconv.FormatInt(d.Size(), 36)
+	return fmt.Sprintf(`"%s%s"`, t, s)
+}
```

Comment:
I didn't see the quotes :smile:

Not that it matters I guess, but I did a quick benchmark:

```go
func BenchmarkSprintf(b *testing.B) {
	for n := 0; n < b.N; n++ {
		fmt.Sprintf(`"%s%s"`, s1, s2)
	}
}
func BenchmarkConcat(b *testing.B) {
	for n := 0; n < b.N; n++ {
		_ = `"` + s1 + s2 + `"`
	}
}
```

Results:

```
BenchmarkSprintf-8       5000000      267 ns/op
BenchmarkConcat-8       30000000      42.7 ns/op
```

I'll leave it up to you to decide if we should change Sprintf to `+`. :) Let me know when you're ready for me to merge it in!